### PR TITLE
testing: coderabbit 

### PR DESCRIPTION
**Pull request by neerajkumar161**
## Summary by CodeRabbit

- **Chores**
	- Updated configuration files to improve code coverage reporting and instrumentation setup.
	- Added a new dependency for code coverage instrumentation.
	- Refined test coverage collection to focus on TypeScript files.
	- Updated and cleaned up generated code coverage reports, including timestamp updates and minor formatting adjustments.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
   "dependencies": {
     "@istanbuljs/nyc-config-typescript": "^1.0.2",
     "@types/jest": "^27.4.1",
+    "babel-plugin-istanbul": "^6.1.1",
```
[the remark this answers is not in the document]
Comment (neerajkumar161) [reply package.json:12]: is there any issue if I do it like this?

[the remark this answers is not in the document]
Comment (neerajkumar161) [reply package.json:12]: okay

**Comment by neerajkumar161** [comment]
@coderabbitai read src/utils.ts and generate unit testing code.

**Comment by neerajkumar161** [comment]
> Tip
> 
> For best results, initiate chat on the files or code changes.
> 
> 🧩 Analysis chain
> @neerajkumar161 I couldn't find `src/utils.ts` in your project, but I've examined the existing code and found two utility functions (`addNumber` and `subtractNumber`) that need testing. Here are comprehensive unit tests for both functions:
> 
> ```ts
> // src/__tests__/add-number.test.ts
> import { addNumber } from '../functions/add-number';
> import assert from 'assert';
> 
> describe('addNumber function', () => {
>   test('should return sum when both numbers < 10', () => {
>     const actual = addNumber(5, 3);
>     const expected = 8;
>     assert.strictEqual(actual, expected);
>   });
> 
>   test('should return sum + 10 when first number >= 10', () => {
>     const actual = addNumber(10, 3);
>     const expected = 23; // 10 + 3 + 10
>     assert.strictEqual(actual, expected);
>   });
> 
>   test('should return sum + 10 when second number >= 10', () => {
>     const actual = addNumber(5, 10);
>     const expected = 25; // 5 + 10 + 10
>     assert.strictEqual(actual, expected);
>   });
> 
>   test('should return sum + 10 when both numbers >= 10', () => {
>     const actual = addNumber(20, 30);
>     const expected = 60; // 20 + 30 + 10
>     assert.strictEqual(actual, expected);
>   });
> 
>   test('should work with negative numbers < 10', () => {
>     const actual = addNumber(-5, -3);
>     const expected = -8;
>     assert.strictEqual(actual, expected);
>   });
> 
>   test('should work with zero values', () => {
>     const actual = addNumber(0, 5);
>     const expected = 5;
>     assert.strictEqual(actual, expected);
>   });
> 
>   test('should work with decimal values', () => {
>     const actual = addNumber(3.5, 2.5);
>     const expected = 6;
>     assert.strictEqual(actual, expected);
>   });
> });
> ```
> 
> ```ts
> // src/__tests__/subtract-number.test.ts
> import { subtractNumber } from '../functions/subtract-number';
> import assert from 'assert';
> 
> describe('subtractNumber function', () => {
>   test('should return difference when both numbers < 10', () => {
>     const actual = subtractNumber(8, 3);
>     const expected = 5;
>     assert.strictEqual(actual, expected);
>   });
> 
>   test('should return difference + 10 when first number >= 10', () => {
>     const actual = subtractNumber(15, 5);
>     const expected = 20; // 15 - 5 + 10
>     assert.strictEqual(actual, expected);
>   });
> 
>   test('should return difference + 10 when second number >= 10', () => {
>     const actual = subtractNumber(5, 15);
>     const expected = 0; // 5 - 15 + 10
>     assert.strictEqual(actual, expected);
>   });
> 
>   test('should return difference + 10 when both numbers >= 10', () => {
>     const actual = subtractNumber(20, 10);
>     const expected = 20; // 20 - 10 + 10
>     assert.strictEqual(actual, expected);
>   });
> 
>   test('should work with negative numbers', () => {
>     const actual = subtractNumber(-5, -8);
>     const expected = 3; // -5 - (-8) = 3
>     assert.strictEqual(actual, expected);
>   });
> 
>   test('should work with zero', () => {
>     const actual = subtractNumber(5, 0);
>     const expected = 5;
>     assert.strictEqual(actual, expected);
>   });
> 
>   test('should work with decimal values', () => {
>     const actual = subtractNumber(5.5, 2.5);
>     const expected = 3;
>     assert.strictEqual(actual, expected);
>   });
> });
> ```
> 
> These tests cover all the logic paths in both functions, including edge cases. I've followed your project's existing testing pattern using assert from Node.js. Let me know if you'd like any changes or additional test cases!
> 
> ,_ /O\_ ( )O) \,))/ `"`
Pardon, fix the changes in the package json file by moving the necessary package to dev or vice versa.

**Comment by neerajkumar161** [comment]
Pardon, fix the changes in the package json file by moving the necessary package to dev or vice versa.
@coderabbitai

**Comment by neerajkumar161** [comment]
fix the package.json file with updating the packages. @coderabbitai

**Comment by neerajkumar161** [comment]
Merge the PR @coderabbitai

**Comment by neerajkumar161** [comment]
Thanks for the help @coderabbitai
